Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/ChangeEvent.swift

```diff
@@ -0,0 +1,168 @@
+        var fullDocument: T?
+        if let actualFullDocument = changeEvent.fullDocument {
+            fullDocument = try! BSONDecoder().decode(T.self,
+                                                     from: actualFullDocument)
+        }
+        return ChangeEvent<T>.init(
+            id: changeEvent.id,
+            operationType: changeEvent.operationType,
+            fullDocument: fullDocument,
+            ns: changeEvent.ns,
+            documentKey: changeEvent.documentKey,
+            updateDescription: changeEvent.updateDescription,
+            hasUncommittedWrites: changeEvent.hasUncommittedWrites)
+    }
+
+    /**
+     Generates a change event for a local insert of the given document in the given namespace.
+
+     - parameter namespace: the namespace where the document was inserted.
+     - parameter document: the document that was inserted.
+     - returns a change event for a local insert of the given document in the given namespace.
+     */
+    static func changeEventForLocalInsert(namespace: MongoNamespace,
+                                          document: Document,
+                                          writePending: Bool) -> ChangeEvent<Document> {
+        let docId = document["_id"]
+        return ChangeEvent<Document>.init(
+            id: Document(),
+            operationType: .insert,
+            fullDocument: document,
+            ns: namespace,
+            documentKey: ["_id": docId],
+            updateDescription: nil,
+            hasUncommittedWrites: writePending)
+    }
+
+    /**
+     Generates a change event for a local update of a document in the given namespace referring
+     to the given document _id.
+
+     - parameter namespace: the namespace where the document was inserted.
+     - parameter documentId: the _id of the document that was updated.
+     - parameter update: the update specifier.
+     - returns: a change event for a local update of a document in the given namespace referring
+     to the given document _id.
+     */
+    static func changeEventForLocalUpdate(namespace: MongoNamespace,
+                                          documentId: BSONValue,
+                                          update: UpdateDescription,
+                                          fullDocumentAfterUpdate: Document,
+                                          writePending: Bool) -> ChangeEvent<Document> {
+        return ChangeEvent<Document>(
+            id: Document(),
+            operationType: .update,
+            fullDocument: fullDocumentAfterUpdate,
+            ns: namespace,
+            documentKey: ["_id": documentId],
+            updateDescription: update,
+            hasUncommittedWrites: writePending)
+    }
+
+    /**
+     Generates a change event for a local replacement of a document in the given namespace referring
+     to the given document _id.
+
+     - parameter namespace: the namespace where the document was inserted.
```

Comment:
[nit] replaced

Reply:
+1, should be [typo] and should be fixed